colossus-metrics/src/main/scala/colossus/metrics/Aggregation.scala

```diff
@@ -38,20 +38,23 @@ object TagSelector {
 }
 
 sealed trait AggregationType {
-  def aggregate(values: Seq[Long]): Long
+  def aggregate(values: Seq[MetricValue]): RawMetricValue
 }
 object AggregationType {
   case object Sum extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.sum
+    def aggregate(values: Seq[MetricValue]): RawMetricValue = values.foldLeft(0L){case (build: RawMetricValue, next: MetricValue) => build + next.value}
   }
   case object Average extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.sum / values.size
+    def aggregate(values: Seq[MetricValue]) = values.foldLeft(0L){case (build, next) => build + next.value} / values.size
   }
   case object Max extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.max
+    def aggregate(values: Seq[MetricValue]) = values.foldLeft(0L){case (max, next) => if (max > next.value) max else next.value}
```

Comment:
For both this and Min...if we are iterating over a set of negative values, wouldn't starting at 0 cause some issues?
Should we use reduceLeft instead, and use an empty check to return a default value?